src/output.rs

```diff
@@ -65,7 +67,12 @@ pub fn write_details(entries: &[(DirEntry, Option<Status>)], dir: &Directory, ap
     print!(app, "total ", blocks, "\n");
 
     let mut buf = Buffer::new();
-    largest_size = buf.format(largest_size as u64).len();
+    // Humanized sizes are max len 4: e.g. "3.1M" or "256K"
+    largest_size = if app.humanize {
+        4
```

Comment:
This deserves additional clarification that this assumes that we won't exceed more than 999 TB. Exceeding that value is easy (if you know the right incantation):
```
╭ ➜ ben@archlinux:/tmp
╰ ➤ touch test
╭ ➜ ben@archlinux:/tmp
╰ ➤ truncate -s 1125899906842624 test 
╭ ➜ ben@archlinux:/tmp
╰ ➤ du -h test
0	test
╭ ➜ ben@archlinux:/tmp
╰ ➤ ls -l test
-rw-r--r-- 1 ben ben 1125899906842624 Apr 10 18:12 test
╭ ➜ ben@archlinux:/tmp
╰ ➤ ls -lh test
-rw-r--r-- 1 ben ben 1.0P Apr 10 18:12 test
```

Another way around this is to extend support up to Exabytes. 1 ZB is more than u64::MAX, so that's likely to be unsupported for other reasons.